### PR TITLE
feat(sdk): deprecate CreateQuestion and migrate cli and docs to use InteractiveQuestion

### LINKED ISSUE #49715: Deprecate CreateQuestion and migrate cli and docs to InteractiveQuestion
We can deprecate the CreateQuestion component now that the InteractiveQuestion component supports creating questions.

- We should add a code sample on how to do this in the docs as well, as it can be a little tricky to implement by yourself.
- We should also update the CLI to use InteractiveQuestion.

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/49715

Deprecates the `CreateQuestion` component. Migrates documentation and the Embedding CLI to use `InteractiveQuestion` instead.

# CLI Demo

CleanShot 2567-11-13 at 04 00 19@2x

CleanShot 2567-11-13 at 04 00 34@2x

CleanShot 2567-11-13 at 04 00 25@2x

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import type { InteractiveQuestionProps } from "../InteractiveQuestion";
 
 type CreateQuestionProps = Omit<InteractiveQuestionProps, "questionId">;
 
+/** @deprecated Use `InteractiveQuestion` with `isSaveEnabled={true}` and without specifying `questionId` instead. */
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx:7]: Should we be more aggressive? I'm not sure all IDEs even show a warning for `@deprecated`
Maybe rendering something in the UI?

File: docs/embedding/sdk/questions.md [hunk 2]
```diff
@@ -248,10 +248,68 @@ import {MetabaseProvider, CreateQuestion} from "@metabase/embedding-sdk-react";
...
-            <CreateQuestion/>
+            <InteractiveQuestion
+                questionId={questionId}
+                onSave={(question) => {
+                    setQuestionId(question.id())
+                    setSaveModalOpen(false)
+                }}
+                isSaveEnabled
+            >
+                <div>
```
[the remark this answers is not in the document]
Comment (heypoom) [reply docs/embedding/sdk/questions.md:268]: @deniskaber That's true... I can think of two ways to provide a sensible default for creating questions.

1. Create a `CreateQuestionView` component that provides the default view for creating questions. The downside is the save modal & question state as well as the save handler remains in the snippet. The upside is the snippet is _a tiny bit_ shorter, and we don't have to maintain a separate component:

```tsx
const [questionId, setQuestionId] = useState(undefined)
const [isModalOpen, setModalOpen] = useState(false)

const onSave = (question) => {
  setQuestionId(question.id())
  setModalOpen(false)
}

<InteractiveQuestion questionId={questionId} onSave={onSave} isSaveEnabled>
  <CreateQuestionView isModalOpen={isModalOpen} setModalOpen={setModalOpen} />
</InteractiveQuestion>
```

2. Don't deprecate the `CreateQuestion` component, but change its implementation to directly call `InteractiveQuestion` underneath with a default create question view. We can get rid of `QuestionEditor` and other related components. The downside is you have to maintain it as an alternative to `InteractiveQuestion`.

cc @oisincoveney @npretto what do you think?

[the remark this answers is not in the document]
Comment (WiNloSt) [reply docs/embedding/sdk/questions.md:268]: I like the goal of having a sensible default, your example isn't entirely bad, but it's for after users using a default component and want more customization.

What you have here could be their own section maybe like how to customize `CreateQuestion`.

And I like your 2). I wouldn't want my option to just have one line for creating questions taken away and be replaced with 50-60 more lines of code.

**Review by deniskaber — APPROVED** [review]

**Comment by heypoom** [comment]
This PR is superseded by https://github.com/metabase/metabase/issues/50082

**Pull request closed without merging**